components/scream/src/physics/shoc/shoc_diag_second_moments_lbycond_impl.hpp

```diff
@@ -0,0 +1,48 @@
+
+#ifndef SHOC_DIAG_SECOND_MOMENTS_LBYCOND_IMPL_HPP
+#define SHOC_DIAG_SECOND_MOMENTS_LBYCOND_IMPL_HPP
+
+#include "shoc_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {
+namespace shoc {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::shoc_diag_second_moments_lbycond(
+    const Scalar& wthl_sfc, const Scalar& wqw_sfc, const Scalar& uw_sfc, const Scalar& vw_sfc,
+    const Scalar& ustar2, const Scalar& wstar,
+    Scalar& wthl_sec, Scalar& wqw_sec, Scalar& uw_sec, Scalar& vw_sec,
+    Scalar& wtke_sec, Scalar& thl_sec, Scalar& qw_sec, Scalar& qwthl_sec)
+{
+  // Purpose of this subroutine is to diagnose the lower
+  // boundary condition for the second order moments needed
+  // for the SHOC parameterization.
+  // The thermodymnamic, tracer, and momentum fluxes are set
+  // to the surface fluxes for the host model, while the
+  // thermodynamic variances and covariances are computed
+  // according to that of Andre et al. 1978.
+
+  const auto ufmin = 0.01;
+
+  auto uf = std::sqrt(ustar2+sp(0.3)*wstar*wstar);
+  uf = std::max(ufmin,uf);
```

Comment:
This could be the cause of the segfault. `std::max` is undefined on device, or at least has been in past Cuda versions. Try `ekat::impl::max` instead.

Reply:
Yes, I will replace it with ekat::impl::max<Scalar> function.